sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -149,7 +190,7 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
      * `::AzureSdkGetCustomHttpTransport()` must be linked in the end-user application.
      *
      */
-    std::shared_ptr<HttpTransport> Transport = _detail::GetTransportAdapter();
+    std::shared_ptr<HttpTransport> Transport;
```

Comment:
Theoretically it is a breaking change, but in practice it isn't. Before this change, `Transport` was empty when the customer created a `TransportOptions` object, afterwards it is not. However, if you called into the pipeline with an empty `Transport`, the pipeline constructor called `GetTransportAdapter()`. So functionally these are equivalent. The only difference is that if a customer IS creating their own transport, then there might be a platform default transport provided.

Reply:
1. Would this work with customer-provided `AzureSdkGetCustomHttpTransport()` - our SDK's way for customers to override transport?

2. Why did you need to change it? Is there no other way around it?

Reply:
Yes, it would work with any custom transport - the only difference is that if you built the system with BUILD_ENABLE_CUSTOM_TRANSPORT, your transport object would be constructed slightly later (when the pipeline is constructed, not when the Options object is instantiated) (note: I got it backwards in the earlier comment: before the transport was set before the pipeoline was created, now it's set when the pipeline is created).

And I needed to change it because the old code created the transport object too early - if I wanted to create the transport object in the TransportPolicy, I had to wait until after the customer had a chance to set the various options.